detector/calorimeter/SCEPCal_MainLayer.cpp

```diff
@@ -0,0 +1,690 @@
+  std::cout << "N_THETA_BARREL:       " << N_THETA_BARREL           << std::endl;
+  std::cout << "N_GAMMA_BARREL:       " << N_GAMMA_BARREL           << std::endl;
+  std::cout                                                         << std::endl;
+  std::cout << "N_THETA_ENDCAP:       " << N_THETA_ENDCAP           << std::endl;
+  std::cout << "ENDCAP_THETA_START:   " << ENDCAP_THETA_START       << std::endl;
+  std::cout                                                         << std::endl;
+  std::cout                                                         << std::endl;
+
+  dd4hep::DetElement ScepcalDetElement(detName,detectorXML.id());
+  dd4hep::Volume experimentalHall=theDetector.pickMotherVolume(ScepcalDetElement);
+  dd4hep::xml::Dimension sdType=detectorXML.child(_Unicode(sensitive));
+  sens.setType(sdType.typeStr());
+  dd4hep::Readout readout=sens.readout();
+  dd4hep::Segmentation geomseg=readout.segmentation();
+  dd4hep::Segmentation* _geoSeg=&geomseg;
+  auto segmentation=dynamic_cast<dd4hep::DDSegmentation::SCEPCal_MainSegmentation_k4geo *>(_geoSeg->segmentation());
+  
+  dd4hep::OpticalSurfaceManager surfMgr = theDetector.surfaceManager();
+  dd4hep::OpticalSurface PbWO4_to_ESR = surfMgr.opticalSurface("/world/"+detName+"#PbWO4_to_ESR");
+  
+  std::vector<double> zBarrelPolyhedra   ={-br_phislice_8pa_y2,-br_phislice_8pa_y0,br_phislice_8pa_y0,br_phislice_8pa_y2};
+  std::vector<double> rminBarrelPolyhedra={br_phislice_8pa_z2,br_phislice_8pa_z1,br_phislice_8pa_z1,br_phislice_8pa_z2};
+  std::vector<double> rmaxBarrelPolyhedra={br_phislice_8pa_z2,br_phislice_8pa_z2,br_phislice_8pa_z2,br_phislice_8pa_z2};
+  
+  std::vector<double> zEndcapPolyhedra   ={ec_phislice_8pa_z1,ec_phislice_8pa_z2};
+  std::vector<double> rminEndcapPolyhedra={ec_beg_phislice_8pa_y0,ec_beg_phislice_8pa_y2};
+  std::vector<double> rmaxEndcapPolyhedra={ec_end_phislice_8pa_y0,ec_end_phislice_8pa_y2};
+
+  std::vector<double> zEndcapPolyhedra_1   ={-ec_phislice_8pa_z2,-ec_phislice_8pa_z1};
+  std::vector<double> rminEndcapPolyhedra_1={ec_beg_phislice_8pa_y2,ec_beg_phislice_8pa_y0};
+  std::vector<double> rmaxEndcapPolyhedra_1={ec_end_phislice_8pa_y2,ec_end_phislice_8pa_y0};
+
+  auto CreatePolyhedraShapeVolume_SetVolAttributes_Place_SetCellId = [
+    &theDetector, 
+    &segmentation
+  ](
+    std::string volName,
+    int nSegments,
+    double phiOffset,
+    double phiExtent,
+    std::vector<double> zPoints,
+    std::vector<double> rMin,
+    std::vector<double> rMax,
+    xml_comp_t compXml,
+    dd4hep::Volume assemblyVol,
+    dd4hep::PlacedVolume *thePlacedVol=nullptr
+  ) {
+      dd4hep::Polyhedra theShape(nSegments,phiOffset,phiExtent,zPoints,rMin,rMax);
+      dd4hep::Volume theVolume(volName, theShape, theDetector.material("Vacuum"));
+      theVolume.setVisAttributes(theDetector,compXml.visStr());
+      *thePlacedVol=assemblyVol.placeVolume(theVolume);
+      return theVolume;
+  };
```

Comment:
I'm not sure how `dd4hep::Volume` will behave in the `return`... Is this lambda function really necessary?